cdap-source-control/src/main/java/io/cdap/cdap/sourcecontrol/operationrunner/InMemorySourceControlOperationRunner.java

```diff
@@ -144,21 +144,13 @@ private Path validateAppConfigRelativePath(RepositoryManager repositoryManager,
                                              Path appRelativePath)
     throws PushFailureException {
     Path filePath = repositoryManager.getRepositoryRoot().resolve(appRelativePath);
-    try {
-      if (Files.exists(filePath)) {
-        Path resolvedPath = filePath.toRealPath();
-        Path resolvedRepoRootPath = repositoryManager.getRepositoryRoot().toRealPath();
-        if (!resolvedPath.startsWith(resolvedRepoRootPath)) {
-          throw new PushFailureException
-            (String.format("Config file %s pointing to invalid path %s", appRelativePath, resolvedPath));
-        }
-        if (!Files.isRegularFile(resolvedPath)) {
-          throw new PushFailureException
-            (String.format("Config file %s is not a regular file", appRelativePath));
-        }
-      }
-    } catch (IOException e) {
-      throw new PushFailureException(String.format("Failed to resolve config file path %s", appRelativePath));
+    if (Files.isSymbolicLink(filePath)) {
+      throw new PushFailureException
+        (String.format("Config file %s is a symbolicLink", appRelativePath));
```

Comment:
nit: put the ( on the previous line.

Also would be good to have a little more information, like:

"Cannot push %s to the remote repository because it already exists as a symbolic link. Symbolic links are not allowed."

Reply:
updated